flink-table-store-hive/flink-table-store-hive-catalog/src/test/java/org/apache/flink/table/store/hive/HiveCatalogITCase.java

```diff
@@ -276,6 +279,36 @@ public void testFlinkWriteAndHiveRead() throws Exception {
         }
     }
 
+    @Test
+    public void testRenameTable() throws Exception {
+        tEnv.executeSql("CREATE TABLE t1 (a INT)").await();
+        tEnv.executeSql("CREATE TABLE t2 (a INT)").await();
+        tEnv.executeSql("INSERT INTO t1 VALUES(1)");
+
+        // the source table do not exist.
+        assertThatThrownBy(() -> tEnv.executeSql("ALTER TABLE t3 RENAME TO t4"))
+                .hasMessage(
+                        "Table `my_hive`.`test_db`.`t3` doesn't exist or is a temporary table.");
+
+        // the target table has existed.
+        assertThatThrownBy(() -> tEnv.executeSql("ALTER TABLE t1 RENAME TO t2"))
+                .hasMessage(
+                        "Could not execute ALTER TABLE my_hive.test_db.t1 RENAME TO my_hive.test_db.t2");
+
+        tEnv.executeSql("ALTER TABLE t1 RENAME TO t3").await();
+        List<String> tables = hiveShell.executeQuery("SHOW TABLES");
+        Assert.assertTrue(tables.contains("t3"));
+        Assert.assertFalse(tables.contains("t1"));
+
```

Comment:
Add `tEnv.executeSql("SELECT * FROM t3").print();` here.
You will find the exception.

Reply:
I add the  query test code before ,but get an exception, because the test framework do not rename the location, if we add the query test code, I think others may be confused. They would think that data cannot be read after rename

Reply:
I am wonder that if we can just do the renaming. The hive will rename again. Is this safe?

Reply:
yes ,I think we  only need call hive method `IMetaStoreClient#alter_table` by thrift client, the rest of the work is done by hive ，if we rename the location dir  first, the hive will throw an exception

https://github.com/apache/hive/blob/branch-2.3/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java#L234

Reply:
Maybe we can try to rename location after altering to hive metastore?

Reply:
> Maybe we can try to rename location after altering to hive metastore?

  I don't recommend this

- The hive engine has done this operation. and there is no any problem, we don't need to repeat it 

- This is a bug of HiveRunner, I test the lastest verion(6.0.1), I fount it is fine, the location is updated 
correct，but if  we upgrade Hiverunner to lastest version, some test cases will fail, so I think we should add a todo issue to upgrade HiveRunner instead of manually renaming the location

what do you think about ?

Reply:
Sounds good

Reply:
> Sounds good

sorry for the late update, could you help review it again? thanks